cccatalog-api/test/v1_integration_test.py

```diff
@@ -0,0 +1,502 @@
+            verify=False
+        ).text
+    )
+    # Did quotation marks actually narrow down the search?
+    assert len(no_quotes['results']) > len(quotes['results'])
+    # Did we find only Claude Monet works, or did his lesser known brother Jim
+    # Monet sneak into the results?
+    for result in quotes['results']:
+        assert 'Claude Monet' in result['creator']
+
+
+@pytest.fixture
+def test_oauth2_registration():
+    payload = {
+        'name': 'INTEGRATION TEST APPLICATION {}'.format(uuid.uuid4()),
+        'description': 'A key for testing the OAuth2 registration process.',
+        'email': 'example@example.org'
+    }
+    response = requests.post(
+        API_URL + '/v1/auth_tokens/register', json=payload, verify=False
+    )
+    parsed_response = json.loads(response.text)
+    assert response.status_code == 201
+    return parsed_response
+
+
+@pytest.fixture
+def test_oauth2_token(test_oauth2_registration):
+    client_id = test_oauth2_registration['client_id']
+    client_secret = test_oauth2_registration['client_secret']
+    token_exchange_request = \
+        'client_id={_id}&client_secret={secret}&grant_type=client_credentials' \
+        .format(_id=client_id, secret=client_secret)
+    headers = {
+        'content-type': "application/x-www-form-urlencoded",
+        'cache-control': "no-cache",
+    }
+    response = json.loads(
+        requests.post(
+            API_URL + '/v1/auth_tokens/token/',
+            data=token_exchange_request,
+            headers=headers,
+            verify=False
+        ).text
+    )
+    assert 'access_token' in response
+    return response
+
+
+def test_oauth2_rate_limit_reporting(test_oauth2_token, verified=False):
+    # We're anonymous still, so we need to wait a second before exchanging
+    # the token.
+    time.sleep(1)
+    token = test_oauth2_token['access_token']
+    headers = {
+        'Authorization': f'Bearer {token}'
+    }
+    response = json.loads(
+        requests.get(f'{API_URL}/v1/rate_limit', headers=headers).text
+    )
+    if verified:
+        assert response['rate_limit_model'] == 'standard'
+        assert response['verified'] is True
+    else:
+        assert response['rate_limit_model'] == 'standard'
+        assert response['verified'] is False
+
+
+@pytest.fixture(scope='session')
+def django_db_setup():
+    if API_URL == 'http://localhost:8000':
+        cccatalog.settings.DATABASES['default'] = {
+            'ENGINE': 'django.db.backends.postgresql',
+            'HOST': '127.0.0.1',
+            'NAME': 'openledger',
+            'PASSWORD': 'deploy',
+            'USER': 'deploy',
+            'PORT': 5432
+        }
+
+
+@pytest.mark.django_db
+def test_oauth2_verification(test_oauth2_token, django_db_setup):
+        # This test needs to cheat by looking in the database, so it will be
+        # skipped in non-local environments.
+        if API_URL == 'http://localhost:8000':
```

Comment:
we should really use DRF's built in `APITestCase` and `APILiveServerTestCase` for all these tests to make the code cleaner and reduce repeated code like defining API URLs.

That can be a future refactor, though, it doesn't have to block this PR.

Reply:
I'm going to resolve this since there aren't any action items for this PR.